app/backplane/power_module/src/c_sensing_tenant.cpp

```diff
@@ -0,0 +1,67 @@
+#include "c_sensing_tenant.h"
+#include "c_power_module.h"
+
+#include <f_core/device/sensor/c_shunt.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(CSensingTenant);
+
+extern k_msgq broadcastQueue;
+
+void CSensingTenant::Startup() {
+}
+
+void CSensingTenant::PostStartup() {
+}
+
+void CSensingTenant::Run() {
+    // TODO: Zero out when we have simulated shunt
+    CPowerModule::SensorData data{
+        .RailBattery = {
+            .Voltage = 12.0f,
+            .Current = 0.0f,
+            .Power = 0.0f
+        },
+        .Rail3v3 = {
+            .Voltage = 3.3f,
+            .Current = 0.0f,
+            .Power = 0.0f
+        },
+        .Rail5v0 = {
+            .Voltage = 5.0f,
+            .Current = 0.0f,
+            .Power = 0.0f
+        }
+    };
+
+#ifndef CONFIG_ARCH_POSIX
+    CShunt shuntBatt(*DEVICE_DT_GET(DT_ALIAS(shunt_batt)));
+    CShunt shunt3v3(*DEVICE_DT_GET(DT_ALIAS(shunt_3v3)));
+    CShunt shunt5v0(*DEVICE_DT_GET(DT_ALIAS(shunt_5v0)));
+    CSensorDevice *sensors[] = {&shuntBatt, &shunt3v3, &shunt5v0};
+#endif
+
+    while (true) {
+#ifndef CONFIG_ARCH_POSIX
+        for (auto sensor: sensors) {
+            sensor->UpdateSensorValue();
+        }
+
+        data.Rail3v3.Current = shunt3v3.GetSensorValueFloat(SENSOR_CHAN_CURRENT);
```

Comment:
now that we chilling with subclasses and stuff, do we want to add our own wrappers so we can say
```c++
class CShunt {
    ...
    NTypes::NSensor::Shunt GetAll();
    ...
};
```
then 
```c++
data.Rail3v3 = shunt3v3.GetShunt();
```
we have so much power to make things pleasant now

Reply:
Eventually yes. I actually hate calling multiple at the moment, but I just want the basics right now.

Reply:
Consider it #180